### ISSUE TITLE
chroot.var file method vs environment variable method

### ISSUE THREAD

**Issue by adrelanos** [martincmelik/Securix-Linux#80]
I am wondering if function `f_chroot_variables` and `source /chroot.var` could be simplified.

Why not just `export` all the variables you want to be accessible within chroot?

I think that would simplify the code a bit. What do you think?

**Comment by martincmelik** [comment]
Because when it happen, that you need to access debug mode (for chroot.sh in exit_on_error function), you will loose all variables
As far as I remember even when I export all variables I was facing some other issues in chroot environment

**Comment by adrelanos** [comment]
Variables should not be lost when starting `/bin/bash` from `exit_on_error function`. Child processes should always inherit parent environment variables, unless you are using `sudo` without `-E` (preserve environment). If that happens, you might have found an interesting bug in `bash`.

https://github.com/martincmelik/Securix-Linux/blob/master/securix-install/install.sh#L1115
```
screen chroot /mnt/gentoo/ /bin/bash chroot.sh
```

What is `screen` good for here?

I think you could drop the `/bin/bash` from that line. The shebang in `chroot.sh` (`#!/bin/bash`) should make sure it's executed by bash, no?

**Comment by martincmelik** [comment]
I would expect that as well, but it happen to me. Not during install.sh execution, but in chroot environment.

screen is a wrapper that allows separation between the text program and the shell from which it was launched. So it will switch your screen to output of chroot execution. Needed for cases when there will be some additional question or action needed from user (e.g. when you're willing to change some options in kernel)

otherwise script will work for couple of minutes/hours and you will not see any output

**Comment by adrelanos** [comment]
> /bin/bash is needed, it wasn't work as you're switching to completely new environment and you need shell

Because /bin/bash isn't installed at that point in chroot? Because otherwise auto run of required interpreter using shebang otherwise works flawless, no?

**Comment by martincmelik** [comment]
Feel free to try it

**Comment by adrelanos** [comment]
> screen is a wrapper that allows separation between the text program and the shell from which it was launched.

But it's not technically necessary?

> So it will switch your screen to output of chroot execution.

I find this rather problematic. Because then the output of the chroot script is not visible in the usual output / log.

> Needed for cases when there will be some additional question or action needed from user (e.g. when you're willing to change some options in kernel)

chroot should have no effect on stdin. This is tested in Whonix's build script, which also has an interactive error handler.

**Comment by martincmelik** [comment]
This code has been invented almost three years ago
If you think that it is not needed anymore, then please try it first

**Comment by adrelanos** [comment]
Yes, need to address this at some point. The screen/log issue is important to Whonix-Gentoo-Hardened project.

**Comment by martincmelik** [comment]
Will modify install.sh and try during next test build

**Comment by martincmelik** [comment]
First tests failed. Some variables (for reasons unknown to me) hasn't been propagated.
Will debug it more later
